### ISSUE TITLE
setMaxLineLength does not work

### ISSUE THREAD

**Issue by JustBlackBird** [swiftmailer/swiftmailer#570]
As I can see `setMaxLineLength` does not actually set length of lines in `text/plain` message. A message is sent correctly but long lines aren't broken. Here is my use case:

``` php
require __DIR__ . "/vendor/autoload.php";

$message = Swift_Message::newInstance()
    ->setContentType('text/plain')
    ->setCharset('utf-8')
    ->setMaxLineLength(85)
    ->setFrom('test@example.com')
    ->setTo('test@example.com')
    ->setReplyTo('test@example.com')
    ->setSubject('A test message')
    ->setBody(str_repeat('This line is too long and must be broken. ', 10));

$mailer = Swift_Mailer::newInstance(Swift_MailTransport::newInstance());
$mailer->send($message);
```

and here is my `composer.json`:

``` json
{
    "require": {
        "swiftmailer/swiftmailer": "5.3.1"
    }
}
```

I'm using php 5.5.17.

Is this a bug or may be I've missed something?

UPD: Forgot to mention there is nothing in Apache's error log

**Comment by X-Bird** [comment]
me too!

**Comment by Alexander--** [comment]
I have hit the same issue. After quick investigation I discovered that stock SwiftMailer configuration does not actually support long line lengths in practice.

Base64 encoder does not take advantage of them because SwiftMailer follows RFC, and RFC explicitly prohibits lines over 76 bytes long.

Quoted-printable encoder does not take advantage of long lines for the same reason.

8bit encoder *might* take advantage of long lines, but it requires considerable persistence to use in multipart messages because of [this line](https://github.com/swiftmailer/swiftmailer/blob/85b7e7d12e2d8d897bdba1e0a6496e2c6d700164/lib/classes/Swift/Mime/SimpleMimeEntity.php#L735) in SimpleMimeEntity.

TL; DR: the only way to use long lines (or custom encoders) with SwiftMailer is by using a custom message class registered via `Swift_DependencyContainer`, but that line causes nested parts of multipart message to be created using base class instead of registered custom class… So you need to recursively set line lengths on every child whenever you add a part to multipart message.

**Comment by Alexander--** [comment]
For reference, here is how I am currently crafting a long-line HTML message:

```php
<?php

namespace App\Mail;

use Swift_MimePart;

class CleverSwiftMessage extends \Swift_Message
{
    const LINE_LENGTH = 400;

    private $plaintextEncoder;
    private $htmlEncoder;
    private $b64Encoder;

    public function __construct($subject = null, $body = null, $contentType = null, $charset = null)
    {
        parent::__construct($subject, $body, $contentType, $charset);

        $this->setMaxLineLength(CleverSwiftMessage::LINE_LENGTH);

        if ($contentType == 'text/plain') {
            $encoder = $this->getPlaintextEncoder();
        } else if ($contentType == 'text/html') {
            $encoder = $this->getHtmlEncoder();
        } else {
            $encoder = $this->getB64Encoder();
        }

        $this->setEncoder($encoder);
    }

    public function setBody($body, $contentType = null, $charset = null)
    {
        if ($contentType == 'text/plain') {
            $encoder = $this->getPlaintextEncoder();
        } else if ($contentType == 'text/html') {
            $encoder = $this->getHtmlEncoder();
        } else {
            $encoder = $this->getB64Encoder();
        }

        $this->setEncoder($encoder);

        return parent::setBody($body, $contentType, $charset);
    }

    public function setChildren(array $children, $compoundLevel = null)
    {
        parent::setChildren($children, $compoundLevel);

        foreach ($this->getChildren() as $child) {
            $child->setMaxLineLength(CleverSwiftMessage::LINE_LENGTH);
        }

        return $this;
    }

    public function addPart($body, $contentType = null, $charset = null)
    {
        if ($contentType == 'text/plain') {
            $encoder = $this->getPlaintextEncoder();
        } else if ($contentType == 'text/html') {
            $encoder = $this->getHtmlEncoder();
        } else {
            $encoder = $this->getB64Encoder();
        }

        return $this->attach((new Swift_MimePart($body, $contentType, $charset))->setEncoder($encoder));
    }

    private function getB64Encoder() {
        if ($this->b64Encoder == null) {
            $this->b64Encoder = new \Swift_Mime_ContentEncoder_Base64ContentEncoder();
        }

        return $this->b64Encoder;
    }

    private function getHtmlEncoder() {
        if ($this->htmlEncoder == null) {
            $this->htmlEncoder = new \Swift_Mime_ContentEncoder_PlainContentEncoder("8bit");
        }

        return $this->htmlEncoder;
    }

    private function getPlaintextEncoder() {
        return $this->getB64Encoder();
    }
}
````

400 symbols is chosen as reasonable compromise between efficiency and compatibility with broken implementations like [MS Exchange](https://serverfault.com/q/289819/428480).

Note, that I am using base64 for our plaintext part (which is mostly non-ASCII utf8 in our emails) because it is too much pain to manually hard-wrap our email templates at safe length I and don't want "helpful" intermediate servers to line-wrap my 8bit plaintext. This is of course not a problem for html part, where line wraps are rendered with `<br>` tags
